sentinel/tests/unit/transitions.js

```diff
@@ -1,304 +1,193 @@
-const sinon = require('sinon'),
-  assert = require('chai').assert,
-  db = require('../../src/db'),
-  transitions = require('../../src/transitions'),
-  config = require('../../src/config'),
-  metadata = require('../../src/lib/metadata'),
-  tombstoneUtils = require('@medic/tombstone-utils');
+const sinon = require('sinon');
+const assert = require('chai').assert;
+const db = require('../../src/db');
+const transitions = require('../../src/transitions');
+const config = require('../../src/config');
+const metadata = require('../../src/lib/metadata');
+const feed = require('../../src/lib/feed');
+const tombstoneUtils = require('@medic/tombstone-utils');
 
 config.initTransitionLib();
 const infodoc = config.getTransitionsLib().infodoc;
 
 describe('transitions', () => {
   afterEach(() => {
     transitions._changeQueue.kill();
-    transitions._detach();
     sinon.restore();
   });
 
-  it('loadTransitions load throws detach is called', () => {
-    const load = sinon.stub(transitions._transitionsLib, 'loadTransitions').throws();
-    const attach = sinon.stub(transitions, '_attach');
-    const detach = sinon.stub(transitions, '_detach');
-    transitions.loadTransitions();
-    assert.equal(load.callCount, 1);
-    assert.equal(attach.callCount, 0);
-    assert.equal(detach.callCount, 1);
-  });
-
-  it('attach handles missing meta data doc', done => {
-    sinon.stub(db.sentinel, 'get').rejects({ status: 404 });
-    sinon.stub(db.medic, 'get').rejects({ status: 404 });
-
-    const insert = sinon.stub(db.sentinel, 'put').resolves({});
+  describe('loadTransitions', () => {
 
-    const on = sinon.stub().returns({ on: () => ({ cancel: () => null }) });
-    const feed = sinon.stub(db.medic, 'changes').returns({ on: on });
-
-    const processChange = sinon.stub(transitions._transitionsLib, 'processChange').callsArg(1);
-    // wait for the queue processor
-    transitions._changeQueue.drain(() => {
-      assert.equal(processChange.callCount, 1);
-      assert.equal(processChange.args[0][0].id, 'abc');
-      assert.equal(processChange.args[0][0].seq, 55);
-      assert.equal(insert.callCount, 1);
-      assert.equal(insert.args[0][0]._id, '_local/sentinel-meta-data');
-      assert.equal(insert.args[0][0].processed_seq, 55);
-      done();
-    });
-    transitions._attach().then(() => {
-      assert.equal(feed.callCount, 1);
-      assert.equal(feed.args[0][0].since, 0);
-      // invoke the change handler
-      on.args[0][1]({ id: 'abc', seq: 55 });
+    it('detach is called when load throws', () => {
```

Comment:
Since there's no more "detach", maybe we should rename this test.

Reply:
Done.